health/health_test.go

```diff
@@ -0,0 +1,354 @@
+package health
+
+import (
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"os"
+	"testing"
+
+	fs "github.com/featureform/filestore"
+	"github.com/featureform/metadata"
+	pc "github.com/featureform/provider/provider_config"
+	pt "github.com/featureform/provider/provider_type"
+	"github.com/joho/godotenv"
+	"go.uber.org/zap/zaptest"
+)
+
+var providerType = flag.String("provider", "", "provider type under test")
+
+type testMember struct {
+	providerDef metadata.ProviderDef
+}
+
+func TestHealth_Check(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test")
+	}
+
+	err := godotenv.Load("../.env")
+	if err != nil {
+		t.Fatal(err)
```

Comment:
This might cause the test to fail in CI/CD because the `.env` file wouldn't exist.

Reply:
+1, for unit testing you can instead use the testing lib's `.Setenv(_ , _)` mock.
unless there's a specific reason we need to load a `.env` file

Reply:
Agreed, but many of our other tests are currently already relying on this pattern, right?